### ISSUE TITLE
Add detailed vulnerability descriptions to the wiki

### ISSUE THREAD

**Issue by JoranHonig** [ConsenSys/mythril#225]
## Description
At the moment the vulnerability analysis modules create Issues, these issues contain some information that describes the found vulnerability. Now, mythril is not always going to be used by security expert, and mythril should provide the necessary information to solve and identify a vulnerability.

## General layout of wiki pages:
- Introduction
- Example vulnerabilities
- Example fixes
- Links to further documentation

**Comment by JoranHonig** [comment]
@chronaeon could you help with this? There are a lot of wiki pages wanting to be written :smile:

**Comment by chronaeon** [comment]
Definitely! Didn't see this message until now. Roughly how many vulnerabilities does Mythril identify currently, so I can get a size of the project?

**Comment by JoranHonig** [comment]
@chronaeon Awesome!. I can setup the skeleton up tomorrow
One doc page I already built is the one for the integer overflow: https://github.com/ConsenSys/mythril/wiki/Integer-Overflow

One thing we should probably watch out for is that we don't do double work, as things like https://consensys.github.io/smart-contract-best-practices/known_attacks already contain lots of valuable information. The main aim of the docs will be to explain why the vulnerability is a vulnerability, how to fix it and where to find more information.

**Comment by chronaeon** [comment]
Sounds good. Small question about the notation in your first article on Integer Overflow.
>The maximum value for an integer is 2^256 - 1, if it reaches this value it overflow and go back to 0.
Does `2^256` represent an exponent? Sorry if this is a stupid question. That's 77 places in decimal, which seems too big, so I want to make sure I'm not confused about what the `^` means in this context. Still learning all this stuff :)

**Comment by norhh** [comment]
@chronaeon  yes ^ means an exponent

**Comment by chronaeon** [comment]
Thanks !

**Comment by rocky** [comment]
I made some changes to [Integer Overflow](https://github.com/ConsenSys/mythril/wiki/Integer-Overflow). Note that the Wiki is in git, and can be checkout out in git separately; so these changes can easily be reverted or you can see changes in detail. Although the git commit messages have some of the reason for the changes, I'll explain below.

In the particular situation we can list the EVM instructions which cause overflow. The specific instructions are `ADD`, `MUL`, and `EXP` and possibly `XADD` and `XMUL`. I think `ADDMON` and `MULMOD` don't. The reason for giving the instructions is that the specific instruction is listed in the error report that Mythril provides.

I think Mythril currently has a limitation that it only detects 2^256 kinds of overflows. If that's the case we should probably mention it, unless we are planning on addressing it soon. Here, I think you need to fold in information from static analysis. Static analysis can list bounds on specific instructions, and I think with that myth can use that information without much effort.

Here are some general markdown tips. If you add "solidity" after 3 backticks and github' markdown will color the code. See for example:

```
function run(unit256) public {
}
```

versus:

```solidity
function run(unit256) public {
}
```

When put the examples in a contract and compiled them, solidity tells me I need "public" or "private" in the function which is missing from the Ethereum examples. Rather than keep fidelity to the example I think it helpful to have code that doesn't compile with warnings.

Given that I was modifying the examples, I also pulled out the URL and made that clickable which I think is preferable. I know these are also listed in the references section at the bottom, but having the link near where that code is discussed I think helpful. Since the bottom section may have links to discussions that don't have examples or examples we use, that section should probably stay. Having duplicate links doesn't bother me.

And while on the references section, I put these in a markdown list rather than one per paragraph. I also added a blank line after section headings which I did not just because it might look nicer in editing, but I think there are some markdown renders that _require_ the blank line.

Feel free to push back on the changes or revert changes.

**Comment by JoranHonig** [comment]
This issue is on hold with the work going on around Omnibus we shouldn't do duplicate work

**Comment by b-mueller** [comment]
Vulnerability descriptions will be moved to the SWC. Can we close this?
